supreme/src/iosMain/kotlin/at/asitplus/signum/supreme/sign/EphemeralKeysImpl.kt

```diff
@@ -79,28 +83,27 @@ internal actual fun makeEphemeralKey(configuration: EphemeralSigningKeyConfigura
             kSecPrivateKeyAttrs mapsTo cfDictionaryOf(kSecAttrIsPermanent to false)
             kSecPublicKeyAttrs mapsTo cfDictionaryOf(kSecAttrIsPermanent to false)
         }
-        val pubkey = alloc<SecKeyRefVar>()
-        val status = SecKeyGeneratePair(attr, pubkey.ptr, key.ptr)
-        if (status != errSecSuccess) {
-            throw CFCryptoOperationFailed(thing = "generate ephemeral key", osStatus = status)
-        }
-        val pubkeyBytes = corecall {
-            SecKeyCopyExternalRepresentation(pubkey.value, error)
-        }.let { it.takeFromCF<NSData>() }.toByteArray()
+        val privateKey = corecall {
+            SecKeyCreateRandomKey(attr, error)
+        }.manage()
+        val pubkeyBytes = SecKeyCopyPublicKey(privateKey.value).also { defer { CFRelease(it) } }
+        .let {
+            corecall {
+                SecKeyCopyExternalRepresentation(it, error)
+            }
+        }.takeFromCF<NSData>().toByteArray()
```

Comment:
`SecKeyGeneratePair` is deprecated, so replacing it with `SecKeyCreateRandomKey` (which gives us a managed key as well)

Reply:
good catch!